File: idakonyvtar/src/test/java/eu/eyan/idakonyvtar/OszkKeresoTest.java
package eu.eyan.idakonyvtar;

import static org.fest.assertions.Assertions.assertThat;

import java.io.IOException;

import org.fest.assertions.Fail;
import org.junit.Test;

import eu.eyan.idakonyvtar.oszk.OszkKereso;

public class OszkKeresoTest
{
    final static String ABIGEL_ISBN = "9789631193701";

    @Test
    public void isbnKeresOszkban_muxik()
    {
        try
        {
            assertThat(OszkKereso.isbnKeresOszkban("9789631193701")).contains("Abigél");
            assertThat(OszkKereso.isbnKeresOszkban("9789633708316")).contains("Királyok");
            assertThat(OszkKereso.isbnKeresOszkban("9789632273822")).contains("Hallgatni");
        }
        catch (IOException e)
        {
            e.printStackTrace();
            Fail.fail(e.getMessage());
        }
    }
}

File: src/test/scala/eu/eyan/idakonyvtar/oszk/OszkKeresoTest.scala
package eu.eyan.idakonyvtar.oszk

import java.io.IOException
import java.util.concurrent.TimeUnit

import eu.eyan.testutil.ScalaEclipseJunitRunner
import org.fest.assertions.Assertions.assertThat
import org.fest.assertions.Fail
import org.junit.{Rule, Test}
import org.junit.rules.Timeout
import org.junit.runner.RunWith

@RunWith(classOf[ScalaEclipseJunitRunner])
class OszkKeresoTest {
  val ABIGEL_ISBN = "9789631193701"

  val TIMEOUT = 15

  @Rule
  def globalTimeout = new Timeout(TIMEOUT, TimeUnit.SECONDS)

  @Test def isbnKeresOszkban_muxik() = {
    try {
      assertThat(OszkKereso.isbnKeresOszkban(ABIGEL_ISBN)).contains("Abigél")
    } catch {
      case e: IOException => e.printStackTrace(); Fail.fail(e.getMessage)
    }
  }

  @Test def marc_parse_muxik() = {
    val abigel = OszkKereso.getMarcsToIsbn(ABIGEL_ISBN)
    assertThat(findMarc(abigel, CIM)).isEqualTo("Abigél")

    val marai = OszkKereso.getMarcsToIsbn("9789632273822")
    assertThat(findMarc(marai, CIM)).isEqualTo("Hallgatni akartam")
  }

  def findMarc(marcs: List[Marc], marcCode: MarcCodes) = {
    marcs.find(marc =>
      marcCode.marc1.equals(marc.marc1)
        && marcCode.marc2.equals(marc.marc2)
        && marcCode.marc3.equals(marc.marc3)).orNull.value
  }

  class MarcCodes(val marc1: String, val marc2: String, val marc3: String)
  case object CIM extends MarcCodes("245", "10", "a")
}